src/main/java/bio/terra/cli/service/ExternalCredentialsManagerService.java

```diff
@@ -0,0 +1,162 @@
+package bio.terra.cli.service;
+
+import bio.terra.cli.businessobject.Context;
+import bio.terra.cli.businessobject.Server;
+import bio.terra.cli.exception.SystemException;
+import bio.terra.cli.service.utils.HttpUtils;
+import bio.terra.externalcreds.api.SshKeyPairApi;
+import bio.terra.externalcreds.client.ApiClient;
+import bio.terra.externalcreds.model.SshKeyPair;
+import bio.terra.externalcreds.model.SshKeyPairType;
+import com.google.auth.oauth2.AccessToken;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.apache.http.util.TextUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.RestTemplate;
+
+public class ExternalCredentialsManagerService {
+  private static final Logger logger = LoggerFactory.getLogger(WorkspaceManagerService.class);
+
+  // the Terra environment where the WSM service lives
+  private final Server server;
+
+  // the client object used for talking to WSM
+  private final ApiClient apiClient;
+
+  private ExternalCredentialsManagerService(@Nullable AccessToken accessToken, Server server) {
+    this.server = server;
+    this.apiClient =
+        new ApiClient(new RestTemplate(List.of(new MappingJackson2HttpMessageConverter())));
+
+    this.apiClient.setBasePath(server.getExternalCredsUri());
+    if (accessToken != null) {
+      // fetch the user access token
+      // this method call will attempt to refresh the token if it's already expired
+      this.apiClient.setAccessToken(accessToken.getTokenValue());
+    }
+  }
+
+  public static ExternalCredentialsManagerService returnFromContext() {
+    return new ExternalCredentialsManagerService(
+        Context.requireUser().getUserAccessToken(), Context.getServer());
+  }
+
+  /**
+   * Gets Ssh key pair from ECM. If one does not exist yet, generate one.
+   *
+   * @param keyPairType git server tied to the key (e.g. GitHub, GitLab, Azure).
+   */
+  public SshKeyPair getSshKeyPair(SshKeyPairType keyPairType) {
```

Comment:
nit: maybe name something like `getOrCreateSshKeyPair`?

Reply:
Jenni suggested to not have a create because the UI doesn't have a create